lib/utils/file_reader.rb

```diff
@@ -0,0 +1,24 @@
+# encoding: utf-8
+# author: ERAMOTO Masaya
+
+module FileReader
+  def read_file_content(path)
+    # these are currently ResourceSkipped to maintain consistency with the resource
+    # pre-refactor (which used skip_resource). These should likely be changed to
+    # ResourceFailed during a major version bump.
+    file = inspec.file(path)
+    if !file.file?
+      raise Inspec::Exceptions::ResourceSkipped, "Can't find file: #{path}"
+    end
+
+    raw_content = file.content
+    if raw_content.nil?
+      raise Inspec::Exceptions::ResourceSkipped, "Can't read file: #{path}"
+    end
+    if raw_content.empty?
```

Comment:
This feels like something that should be left up to each resource. What if a file is expected to be empty?

Reply:
I thought that users expected that any description exists in a file if they specified its path. But they are permitted to read it even if it is empty so they may expected that. Thus I will fix above.

Reply:
One suggestion: leave this here but make it optional. The caller can specify an option, like so:

`read_file_contents(‘/tmp/empty’, allow_empty: true)`

Reply:
Let me know if you like that idea and if you need help implementing it! 🙂

Reply:
I apologize that I was unaware of your helpful advice.
I implemented to split and wrap it but I will rework to implement the caller having the optional argument.

Reply:
I changed `read_file_content` as described later.
The optional argument is set at false as default since I think `allow_empty: true` should be explicitly specified in case allowing to read a empty file.

```Ruby
--- a/lib/utils/file_reader.rb
+++ b/lib/utils/file_reader.rb
@@ -2,7 +2,7 @@
 # author: ERAMOTO Masaya

 module FileReader
-  def read_file_content(path)
+  def read_file_content(path, allow_empty = false)
     # these are currently ResourceSkipped to maintain consistency with the resource
     # pre-refactor (which used skip_resource). These should likely be changed to
     # ResourceFailed during a major version bump.
@@ -15,7 +15,8 @@ module FileReader
     if raw_content.nil?
       raise Inspec::Exceptions::ResourceSkipped, "Can't read file: #{path}"
     end
-    if raw_content.empty?
+
+    if !allow_empty && raw_content.empty?
       raise Inspec::Exceptions::ResourceSkipped, "File is empty: #{path}"
     end

```

By the way, this PR does not allow dh_params and x509_certificate resources to read a empty file since they always raise a exception if a file is empty.